Panic due to nil pointer to type that implements IsZero()

Starting with v0.5.0 of `repr`, some embedded types that implement `IsZero()` can cause a panic if they are `nil`.

I was able to get a failing test case in https://github.com/alecthomas/assert, so sent a PR containing the test case there: https://github.com/alecthomas/assert/pull/25

v0.4.0 of repr allows the test case to pass, and v0.5.x causes it to panic with a stack trace like this:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered, repanicked]
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7d4209]

goroutine 180 [running]:
testing.tRunner.func1.2({0x8e8760, 0xda59c0})
	/opt/hostedtoolcache/go/1.25.4/x64/src/testing/testing.go:1872 +0x237
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.25.4/x64/src/testing/testing.go:1875 +0x35b
panic({0x8e8760?, 0xda59c0?})
	/opt/hostedtoolcache/go/1.25.4/x64/src/runtime/panic.go:783 +0x132
github.com/golang-jwt/jwt/v5.(*NumericDate).IsZero(0xdab300?)
	<autogenerated>:1 +0x9
github.com/alecthomas/repr.(*Printer).reprValue(0xc0004550c0, 0xc000291c10, {0x96dfa0, 0xc000151d20, 0x199}, {0x99011a, 0x2}, 0x1, 0x0)
	/home/runner/go/pkg/mod/github.com/alecthomas/repr@v0.5.2/repr.go:302 +0x1a66
github.com/alecthomas/repr.(*Printer).reprValue(0xc0004550c0, 0xc000291c10, {0x969040, 0xc000151ce0, 0x199}, {0x0, 0x0}, 0x1, 0x0)
	/home/runner/go/pkg/mod/github.com/alecthomas/repr@v0.5.2/repr.go:317 +0x186b
github.com/alecthomas/repr.(*Printer).reprValue(0xc0004550c0, 0xc000291c10, {0x949e00, 0xc000151ce0, 0x16}, {0x0, 0x0}, 0x1, 0x0)
	/home/runner/go/pkg/mod/github.com/alecthomas/repr@v0.5.2/repr.go:357 +0x9cf
github.com/alecthomas/repr.(*Printer).Print(0xc0004550c0, {0xc000291cb0?, 0x1, 0x2?})
	/home/runner/go/pkg/mod/github.com/alecthomas/repr@v0.5.2/repr.go:175 +0xf5
github.com/alecthomas/repr.String({0x949e00, 0xc000151ce0}, {0xc00005c190, 0x1, 0xc000482db0?})
	/home/runner/go/pkg/mod/github.com/alecthomas/repr@v0.5.2/repr.go:402 +0x130
github.com/alecthomas/assert/v2.objectsAreEqual({0x949e00, 0xc000151ce0}, {0x949e00, 0xc000432580}, {0xc000291db0?, 0x2?, 0xc00038c3f0?})
	/home/runner/go/pkg/mod/github.com/alecthomas/assert/v2@v2.11.0/assert.go:349 +0x152
github.com/alecthomas/assert/v2.Equal[...]({0xa5dcd0, 0xc0002b4700}, 0xc000151ce0, 0xc000432580, {0x0, 0x1?, 0x16?})
	/home/runner/go/pkg/mod/github.com/alecthomas/assert/v2@v2.11.0/assert.go:85 +0x365
github.com/uselagoon/ssh-portal/internal/keycloak_test.TestValidateTokenClaims.func1(0xc0002b4700)
	/home/runner/work/lagoon-ssh-portal/lagoon-ssh-portal/internal/keycloak/jwt_test.go:267 +0x605
testing.tRunner(0xc0002b4700, 0xc000454980)
	/opt/hostedtoolcache/go/1.25.4/x64/src/testing/testing.go:1934 +0xea
created by testing.(*T).Run in goroutine 129
	/opt/hostedtoolcache/go/1.25.4/x64/src/testing/testing.go:1997 +0x465
```

Let me know if I can provide more info.

--- a/repr.go
+++ b/repr.go
@@ -299,8 +299,18 @@ func (p *Printer) reprValue(seen map[reflect.Value]bool, v reflect.Value, indent
 					continue
 				}
 
-				if p.omitZero && ((ft.Implements(isZeroerType) && f.CanInterface() && f.Interface().(isZeroer).IsZero()) || f.IsZero()) {
-					continue
+				if p.omitZero {
+					// check if this type is a nil pointer to a type implementing IsZero
+					// with a value receiver and, if so, avoid calling IsZero() on it as
+					// the method call will automatically dereference the nil pointer and
+					// panic.
+					var nilPtrValueReceiver bool
+					if ft.Kind() == reflect.Pointer && f.IsNil() {
+						_, nilPtrValueReceiver = ft.Elem().MethodByName("IsZero")
+					}
+					if (ft.Implements(isZeroerType) && !nilPtrValueReceiver && f.CanInterface() && f.Interface().(isZeroer).IsZero()) || f.IsZero() {
+						continue
+					}
 				}
 
 				if p.omitEmpty && (f.IsZero() ||

--- a/repr_test.go
+++ b/repr_test.go
@@ -81,8 +81,9 @@ func TestReprZeroSliceMapFields(t *testing.T) {
 		B   bool
 		ZC  zeroTest
 		C   zeroTest
-	}{[]string{}, []string{"a", "b"}, map[string]string{}, map[string]string{"a": "b"}, 0, 1, "", "a", false, true, zeroTest{i: 100}, zeroTest{i: 10}}
-	equal(t, `struct { ZSl []string; Sl []string; ZM map[string]string; M map[string]string; ZI int; I int; ZS string; S string; ZB bool; B bool; ZC repr.zeroTest; C repr.zeroTest }{ZSl: []string{}, Sl: []string{"a", "b"}, ZM: map[string]string{}, M: map[string]string{"a": "b"}, I: 1, S: "a", B: true, C: repr.zeroTest{i: 10}}`, String(v, OmitEmpty(false), OmitZero(true)))
+		T   *time.Time
+	}{[]string{}, []string{"a", "b"}, map[string]string{}, map[string]string{"a": "b"}, 0, 1, "", "a", false, true, zeroTest{i: 100}, zeroTest{i: 10}, nil}
+	equal(t, `struct { ZSl []string; Sl []string; ZM map[string]string; M map[string]string; ZI int; I int; ZS string; S string; ZB bool; B bool; ZC repr.zeroTest; C repr.zeroTest; T *time.Time }{ZSl: []string{}, Sl: []string{"a", "b"}, ZM: map[string]string{}, M: map[string]string{"a": "b"}, I: 1, S: "a", B: true, C: repr.zeroTest{i: 10}}`, String(v, OmitEmpty(false), OmitZero(true)))
 }
 
 func TestReprStringArray(t *testing.T) {

